### ISSUE TITLE
Prefer functional notation?

### ISSUE THREAD

**Issue by oskarer** [captbaritone/eslint-plugin-underscore#27]
There are two styles;
**functional notation**
`_.filter(arr, (element) => {...})`

and **object oriented notation**
`_(arr).filter((element) => {...})`

that accomplish the same thing.

I see the benefit of sticking to one of the styles. Would it be wise to have such a rule that enforces one of the styles?

**Comment by captbaritone** [comment]
Good idea. That sounds like a very reasonable rule and it should be pretty easy to implement. Do you have any interest in making a pull request?

**Comment by oskarer** [comment]
Yes, I could give it a try! However I looked in the existing tests and many of them has the object oriented notation as a valid case, here for example: https://github.com/captbaritone/eslint-plugin-underscore/blob/master/tests/lib/rules/no-double-unwrap.js.

Doing a prefer-functional-notation rule would break a lot of those I suppose.

**Comment by captbaritone** [comment]
That should be fine. Each test only tests a single rule so only your tests will be subject to this rule.

## 

Jordan Eldredge

Please excuse any typos or terseness; this email was sent from a mobile device.

**Comment by captbaritone** [comment]
@oskarer Are you still interested in implementing this rule?

**Comment by oskarer** [comment]
Yes sorry for delay, I began work on this a while ago. Think I got the rule working but got stuck on the fixer. Any ideas?

https://github.com/oskarer/eslint-plugin-underscore/commit/ca404d40f1705ffc84c5cd8a8a95a04373df2c76

**Comment by captbaritone** [comment]
Hmm. Not sure about the fixer off the top of my head. Why don't we merge the rule without the fixer first, and add the fixer as a second pull request? I'd be happy to help come up with ideas.

As for the rule name, I think it would be better to have something like `consistent-call-style` with an option to choose between functional or object-oriented.

If you want to make that change, go for it. If not, I'd be happy to use your rule as a starting point and add the options myself.

**Comment by oskarer** [comment]
Yeah, please go ahead! I'm a bit short on time for fixing this atm. And I agree, letting the user choose style makes more sense.

**Comment by captbaritone** [comment]
Cool. When I get a moment, I'll merge in https://github.com/oskarer/eslint-plugin-underscore/commit/ca404d40f1705ffc84c5cd8a8a95a04373df2c76 and tweak it. Thanks for your help!
